src/game/hud.cpp

```diff
@@ -1308,12 +1312,34 @@ namespace hud
 
     void drawpointer(int w, int h, int index)
     {
+        bool use_custom_crosshair_colors = (crosshairweapons & 2) == 0;
         int cs = int((index == POINTER_GUI ? cursorsize : crosshairsize)*hudsize);
         float fade = index == POINTER_GUI ? cursorblend : crosshairblend;
         vec c(1, 1, 1);
         if(game::focus->state == CS_ALIVE && index >= POINTER_HAIR)
         {
-            if(crosshairweapons&2) c = vec::hexcolor(W(game::focus->weapselect, colour));
+            
+            if (use_custom_crosshair_colors)
+            {
+                if (index == POINTER_TEAM)
+                {
+                    c = vec::hexcolor(crosshaircolor_team);
+                } 
+                else if (index == POINTER_ZOOM) // the player is in zoom
+                {
+                    c = vec::hexcolor(crosshaircolor_zoom);
+                }
+                else // default, (support for hit pointer is below)
```

Comment:
Please don't use in-line comments. I don't want to write that next to _each_ of them in your PR.